queries.sql

```diff
@@ -8,3 +8,40 @@ SELECT * FROM animals WHERE weight_kg > 10.5;
 SELECT * FROM animals WHERE neutered = true;
 SELECT * FROM animals WHERE name <> 'Gabumon';
 SELECT * FROM animals WHERE weight_kg>=10.4 AND weight_kg<=17.3;
+
+BEGIN;
+UPDATE animals SET species = 'undefined';
+ROLLBACK;
+
+BEGIN;
+UPDATE animals SET species = 'Digimon' WHERE name LIKE '%mon';
+UPDATE animals SET species = 'Pokemon' WHERE species LIKE '';
+COMMIT;
+
+BEGIN;
+DELETE FROM animals;
+ROLLBACK;
+
+BEGIN;
+DELETE FROM animals WHERE date_of_birth>'01-01-2022';
+SAVEPOINT SP1;
+UPDATE animals SET weight_kg = weight_kg * (-1);
+ROLLBACK TO SP1;
+UPDATE animals SET weight_kg = weight_kg * (-1) WHERE weight_kg<0;
+COMMIT;
+
+SELECT COUNT(*) FROM animals;
+SELECT COUNT(*) FROM animals WHERE escape_attempts = 0;
+SELECT AVG(weight_kg) FROM animals;
+SELECT name FROM animals WHERE escape_attempts = (SELECT MAX(escape_attempts) FROM animals);
+SELECT species, MIN(weight_kg), MAX(weight_kg) FROM animals GROUP BY species;
+SELECT species, AVG(escape_attempts) FROM animals WHERE date_of_birth>'01-01-1990' AND date_of_birth<'01-01-2000' GROUP BY species;
+
+-- Third project
+SELECT name FROM animals JOIN owners ON animals.owner_id = owners.id WHERE owners.full_name = 'Melody Pond';
+SELECT animals.name FROM species JOIN animals ON species.id = animals.species_id WHERE species.name = 'Pokemon';
+SELECT full_name, name FROM animals RIGHT JOIN owners ON animals.owner_id = owners.id;
+SELECT species.name, COUNT(*) FROM species JOIN animals ON species.id = animals.species_id GROUP BY species.name;
+SELECT name FROM animals JOIN owners ON animals.owner_id = owners.id WHERE owners.full_name = 'Jennifer Orwell' AND animals.name LIKE '%mon';
```

Comment:
- [ ] This query is wrong. You are supposed to list all **digimons** owned by **Jennifer Orwell.**. Kindly update it.

![image](https://user-images.githubusercontent.com/86318284/207864394-afba6db5-ab05-4f88-a04a-c35b23ba8098.png)

Reply:
I have made the changes. Thank you!